lxd/cluster/gateway.go

```diff
@@ -177,25 +177,30 @@ func (g *Gateway) HandlerFuncs(nodeRefreshTask func(*APIHeartbeat)) map[string]h
 				return
 			}
 
-			nodes := make([]db.RaftNode, 0)
+			raftNodes := make([]db.RaftNode, 0)
 			for _, node := range heartbeatData.Members {
-				if node.Raft {
-					nodes = append(nodes, db.RaftNode{
-						ID:      node.ID,
+				if node.RaftID > 0 {
```

Comment:
You can't do that. You need the logic I wrote or upon upgrade most nodes will end up with an empty raft_nodes, leading to a broken database.

Reply:
Ah, I see the logic below, sorry :)

Reply:
So it would be a problem if the cluster was to grow in the middle of the upgrade, but hopefully not something we actually need to be concerned about.